extras/src/test/java/com/google/gson/graph/GraphAdapterBuilderTest.java

```diff
@@ -156,10 +201,10 @@ public void testDeserializationWithMultipleTypes() {
     assertThat(company.name).isEqualTo("Google");
     Employee jesse = company.employees.get(0);
     assertThat(jesse.name).isEqualTo("Jesse");
-    assertThat(jesse.company).isEqualTo(company);
+    assertThat(jesse.company).isSameInstanceAs(company);
     Employee joel = company.employees.get(1);
     assertThat(joel.name).isEqualTo("Joel");
-    assertThat(joel.company).isEqualTo(company);
+    assertThat(joel.company).isSameInstanceAs(company);
```

Comment:
Not directly related to the main changes of this pull request, but checking for reference equality here is more correct.